File: nr1-main/queue/videoQueue.js
// BullMQ queue setup for video processing
const { Queue } = require("bullmq");
const IORedis = require("ioredis");

let connection;
try {
  connection = new IORedis(process.env.REDIS_URL || "redis://localhost:6379");
  connection.on('error', (err) => {
    console.warn('⚠️ Redis connection error:', err.message);
  });
} catch (e) {
  console.error('❌ Failed to connect to Redis:', e.message);
  // TODO: Add fallback or degrade gracefully
}

const videoQueue = new Queue("video-processing", { connection });

// TODO: Add retry logic for failed jobs
// TODO: Add monitoring/alerting hooks

module.exports = videoQueue;

File: nr1-main/app/queue/video_queue.py
import redis
from rq import Queue
from ..config import settings

redis_conn = redis.Redis.from_url(settings.REDIS_URL)
video_queue = Queue('video-processing', connection=redis_conn)

# Example enqueue usage:
# job = video_queue.enqueue(process_video_job, args=(...))
